pkg/servicelb/controller.go

```diff
@@ -260,13 +263,68 @@ func (h *handler) podIPs(pods []*core.Pod) ([]string, error) {
 		ips = keys(intIPs)
 	}
 
+	ips, err := filterByIPFamily(ips, svc)
+	if err != nil {
+		return nil, err
+	}
+
 	if len(ips) > 0 && h.rootless {
 		return []string{"127.0.0.1"}, nil
 	}
 
 	return ips, nil
 }
 
+// filterByIPFamily filters ips based on dual-stack parameters of th service
+func filterByIPFamily(ips []string, svc *core.Service) ([]string, error) {
+
+	var ipv4Addresses []string
+	var ipv6Addresses []string
+
+	for _, ip := range ips {
+		if utilsnet.IsIPv4String(ip) {
+			ipv4Addresses = append(ipv4Addresses, ip)
+		}
+		if utilsnet.IsIPv6String(ip) {
+			ipv6Addresses = append(ipv6Addresses, ip)
+		}
+	}
+
+	switch *svc.Spec.IPFamilyPolicy {
+	case "SingleStack":
+		// Check if ipv4 or ipv6
+		if svc.Spec.IPFamilies[0] == "IPv4" {
+			return ipv4Addresses, nil
+		}
+		if svc.Spec.IPFamilies[0] == "IPv6" {
+			return ipv6Addresses, nil
+		}
```

Comment:
Use the upstream consts from `k8s.io/api/core/v1` instead of string literals. For example:

```suggestion
	case corev1.IPFamilyPolicySingleStack:
		// Check if ipv4 or ipv6
		if svc.Spec.IPFamilies[0] == corev1.IPv4Protocol {
			return ipv4Addresses, nil
		}
		if svc.Spec.IPFamilies[0] == corev1.IPv6Protocol {
			return ipv6Addresses, nil
		}
```